doc/can-query-logic.md

```diff
@@ -14,139 +14,141 @@ data caching and real-time behavior.
+  ```js
+  import {QueryLogic} from "can";
+  import {Todo} from "//unpkg.com/can-demo-models@5";
 
-```js
-{
+  const todoQueryLogic = new QueryLogic(Todo);
+  // Perform query logic:
+  const filter = todoQueryLogic.filterMembers({
     // Selects only the todos that match.
     filter: {
-        complete: false
+      complete: false
     },
     // Sort the results of the selection
     sort: "-name",
     // Selects a range of the sorted result
     page: {start: 0, end: 19}
-}
-```
+  },[
+    {id: 1, name: "do dishes", complete: false},
+    {id: 2, name: "mow lawn", complete: true},
+    // ...
+  ]);
+  console.log( filter ); //-> [{id: 1, name: "do dishes", complete: false}]
 
-@param {function|can-reflect/schema} schemaOrType Defines the behavior of
-keys on a [can-query-logic/query]. This is done with either:
-
-  - A constructor function that supports [can-reflect.getSchema can-reflect.getSchema]. Currently, [can-define/map/map] supports the `can.getSchema` symbol:
-    ```js
-    import {DefineMap} from "can";
-
-    const Todo = DefineMap.extend({
-        id: {
-            identity: true,
-            type: "number"
-        },
-        name: "string",
-        complete: "boolean"
-    });
-    new QueryLogic(Todo);
-    ```
-  - A [can-reflect.getSchema schema object] that looks like the following:
-    ```js
-    import {MaybeNumber, MaybeString, MaybeBoolean} from "can"
-    new QueryLogic({
-        // keys that uniquely represent this type
-        identity: ["id"],
-        keys: {
-            id: MaybeNumber,
-            name: MaybeString,
-            complete: MaybeBoolean
-        }
-    })
-    ```
-
-    Note that if a key type (ex: `name: MaybeString`) is __not__ provided, filtering by that
-    key will still work, but there won't be any type coercion. For example, the following
-    might not produce the desired results:
-
-    ```js
-    var queryLogic = new QueryLogic({identity: ["id"]});
-    queryLogic.union(
-        {filter: {age: 7}},
-        {filter: {age: "07"}}) //-> {filter: {age: {$in: [7,"07"]}}}
-    ```
-    Use types like [can-data-types/maybe-number/maybe-number] if you want to add basic
-    type coercion:
-
-    ```js
-    var queryLogic = new QueryLogic({
-        identity: ["id"],
-        keys: {age: MaybeNumber}
-    });
-    queryLogic.union(
-        {filter: {age: 7}},
-        {filter: {age: "07"}}) //-> {filter: {age: 7}}
-    ```
-
-    If you need even more special key behavior, read [defining properties with special logic](#Definingfilterpropertieswithspeciallogic).
+  ```
+  @codepen
+  @highlight 6-15,only
+
+@param {function|can-reflect/schema} schemaOrType Defines the behavior of keys on a [can-query-logic/query]. This is done with either:
+
+- A constructor function that supports [can-reflect.getSchema can-reflect.getSchema]. Currently, [can-define/map/map] supports the `can.getSchema` symbol:
+
+  @sourceref ./can-query-logic-models/todo-example.js
```

Comment:
the indentation is off here.  

![image](https://user-images.githubusercontent.com/78602/46033473-e6e93900-c0c3-11e8-95c7-d34149f1cafe.png)

codepens aren't really necessary if they can't work easily.

Reply:
Indentation is somewhat difficult as the `@codepen` tag breaks a majority of anything beyond `@signatures`.